finalfusion-ffi/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+use std::cell::RefCell;
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::ffi::OsStr;
+use std::fs::File;
+use std::io::BufReader;
+use std::os::raw::c_char;
+use std::os::unix::ffi::OsStrExt;
+use std::ptr;
+
+use finalfusion::prelude::*;
+
+thread_local! {
+    static ERROR: RefCell<CString> = RefCell::new(CString::new(Vec::new()).unwrap());
+}
+
+fn update_error(err: &str) {
+    ERROR
+        .with(|e| e.replace(CString::new(err.as_bytes()).expect("Error string contains nul byte")));
```

Comment:
`CString::new` takes `T: Into<Vec<u8>>`, `.as_bytes` is not necessary.

If we unwrap the CString through `expect`, we should wrap every fallible method in `catch_unwind`.

Reply:
We should not use `catch_unwind`. We have properly maintained `panic`s and assertion errors are for programming errors, so we don't want them to suddenly propagate as errors in the C library. The program should abort (this is why the panic handler is set to abort rather than unwind in `Cargo.toml`).

Reply:
I commented on this below. When I was doing the review the panic handler was not set to abort yet.